scanner/engine/master.cpp

```diff
@@ -460,14 +460,21 @@ grpc::Status MasterImpl::RegisterPythonKernel(
     DeviceType device_type = python_kernel->device_type();
     const std::string& kernel_str = python_kernel->kernel_str();
     const std::string& pickled_config = python_kernel->pickled_config();
+    const int batch_size = python_kernel->batch_size();
     // Create a kernel builder function
     auto constructor = [kernel_str, pickled_config](const KernelConfig& config) {
       return new PythonKernel(config, kernel_str, pickled_config);
     };
     // Create a new kernel factory
     // TODO(apoms): Support batching and # of devices in python kernels
-    KernelFactory* factory =
-        new KernelFactory(op_name, device_type, 1, false, 1, constructor);
+    KernelFactory* factory;
+    if (batch_size > 0) {
+      factory = new KernelFactory(op_name, device_type, 1, true, 
+                                  batch_size, constructor);
+    } else {
+      factory = new KernelFactory(op_name, device_type, 1, false, 
+                                  1, constructor);
```

Comment:
Instead of adding an if statement, I would just add a variable for `can_batch`

```
bool can_batch = (batch_size > 0);
KernelFactory* factory =
  new KernelFactory(op_name, device_type, 1, can_batch, std::max(batch_size, 1), constructor);
```